src/site_assessment/best_fit_polygons.jl

```diff
@@ -54,40 +63,71 @@ function assess_reef_site(
     best_poly = Vector(undef, n_rotations)
     qc_flag = zeros(Int64, n_rotations)
 
-    @floop for (j, r) in enumerate(rotations)
+    for (j, r) in enumerate(rotations)
         rot_geom = rotate_geom(geom, r, target_crs)
         score[j] =
-            size(rel_pix[GO.intersects.([rot_geom], rel_pix.geometry), :], 1) / max_count
+            length(
+                rel_pix[
+                    GO.intersects.([rot_geom], rel_pix.geometry), :lon_idx
+                ]
+            ) / max_count
         best_poly[j] = rot_geom
 
-        if score[j] < surr_threshold
+        if score[j] < suit_threshold
             # Early exit as there's no point in searching further.
             # Changing the rotation is unlikely to improve the score.
             qc_flag[j] = 1
             break
         end
     end
 
-    return score[argmax(score)],
+    return min(score[argmax(score)], 1),
     argmax(score) - (n_per_side + 1),
     best_poly[argmax(score)],
     maximum(qc_flag)
 end
+function assess_reef_site(
+    rel_pix::DataFrame,
+    rotated::Vector{GI.Wrappers.Polygon},
+    max_count::Float64;
+    n_per_side::Int64=1,
+    suit_threshold::Float64=0.7
+)::Tuple{Float64,Int64,GI.Wrappers.Polygon,Int64}
+    # Implementation with pre-rotations
+    n_rotations = length(rotated)
+    score = @MVector zeros(n_rotations)
+    qc_flag = @MVector zeros(Int64, n_rotations)
+
+    for (j, r) in enumerate(rotated)
+        score[j] = count(GO.intersects.([r], rel_pix.geometry)) / max_count
+
+        if score[j] < suit_threshold
+            # Early exit as there's no point in searching further.
+            # Changing the rotation is unlikely to improve the score.
+            qc_flag[j] = 1
+            break
+        end
+    end
+
+    return min(score[argmax(score)], 1),
+    argmax(score) - (n_per_side + 1),
```

Comment:
I think this would read better with an indent, but maybe formatter didn't like it?

Reply:
Auto-format issue - it'll unindent the lines underneath ☹️